pyxem/generators/pdf_generator.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""PDF generator and associated tools.
+
+"""
+
+import numpy as np
+
+from hyperspy.signals import Signal1D
+
+from pyxem.signals.reduced_intensity1d import ReducedIntensity1D
+from pyxem.signals.pair_distribution_function1d import PairDistributionFunction1D
+from pyxem.signals import transfer_navigation_axes
+
+
+class PDFGenerator():
+    """Generates a PairDistributionFunction1D signal from a specified
```

Comment:
Given the PDF could be calculated in more than 1D - do you anticipate this generator being the specialized 1D generator? If so it should probably be PDFGenerator1D() to pre-empt generalization.

Reply:
I imagine this will, if not for simplicity reasons, be better off as a 1D generator. Will change the name.